torch/distributed/_composable/fsdp/fully_shard.py

```diff
@@ -0,0 +1,46 @@
+from typing import Any
+
+import typing_extensions
+
+import torch.nn as nn
+
+from torch.distributed._composable import contract
+from torch.distributed._composable_state import _insert_module_state
+
+
+@contract()
+def fully_shard(
+    module: nn.Module,
+):
+    if isinstance(module, (nn.ModuleList, nn.ModuleDict)):
+        raise ValueError(
+            f"fully_shard does not support containers that do not implement forward: {module}"
+        )
+    state = fully_shard.state(module)
```

Comment:
nit: it took me some time to find out ``fully_shard.state`` is defined in ``@contract``. maybe consider a short comment here to remind people to check ``@contract`` docstrings?